http4s/src/test/scala/io/isomarcte/errors4s/http4s/ResponseErrorProperties.scala

```diff
@@ -0,0 +1,144 @@
+package io.isomarcte.errors4s.http4s
+
+import cats.effect._
+import cats.implicits._
+import fs2._
+import java.nio.charset.StandardCharsets
+import org.http4s._
+import org.http4s.client._
+import org.http4s.laws.discipline.arbitrary._
+import org.scalacheck.Prop.{Status => _, _}
+import org.scalacheck._
+import scodec.bits._
+
+object ResponseErrorProperties extends Properties("ResponseError") {
+  lazy val testRequest: Request[SyncIO] = Request()
+
+  lazy val invalidStatusGen: Gen[Status] = Gen
+    .choose(400, 599)
+    .filterNot(s => s === Status.Gone.code || s === Status.NotFound.code)
+    .map(status => Status(status))
+
+  lazy val nonEmptyStringGen: Gen[String] = Gen.nonEmptyListOf(Arbitrary.arbitrary[Char]).map(_.mkString)
```

Comment:
INFO: Likewise, you could add a `suchThat` clause here and ensure that the simplification would always leave a non-empty string: 
```scala
lazy val nonEmptyStringGen: Gen[String] = 
  Gen.nonEmptyListOf(Arbitrary.arbitrary[Char]).map(_.mkString).suchThat(_.nonEmpty)
```

Reply:
`Gen.nonEmptyListOf(Arbitrary.arbitrary[Char])` is already guaranteed to be non-empty, so I'm going to leave this one.

Reply:
`Gen.nonEmptyListOf` is guaranteed to _generate_ a non-empty list, but any `Shrink` capabilities will ignore that, since at that point, it's just a `String`.